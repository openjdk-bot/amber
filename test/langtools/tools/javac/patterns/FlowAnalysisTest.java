/*
 * @test /nodynamiccopyright/
 * @summary Stress test flow analysis with patterns
 * @compile/fail/ref=FlowAnalysisTest.out -XDrawDiagnostics FlowAnalysisTest.java
 */

class FlowAnalysisTest {
    void foo(Object object) {

        switch (object) {
            case Object o: break;
            case String s: break;
        }
        switch (object) {
            case var o: break;
            case String s: break;
            case "Hello" : break;
        }
    }
}
