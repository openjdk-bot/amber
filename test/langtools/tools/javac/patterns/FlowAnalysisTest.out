FlowAnalysisTest.java:12:13: compiler.err.pattern.dominated
FlowAnalysisTest.java:16:13: compiler.err.pattern.dominated
FlowAnalysisTest.java:17:13: compiler.err.pattern.dominated
3 errors
